### PR TITLE
Allow footer licence customization

### LINKED ISSUE #1059: Footer: No way to create Footer without OGL
Not everyone is going to be using the OGL how can i create a custom message to overwrite the text and link.

### PR DESCRIPTION
**Pull request by domdomegg**
Closes https://github.com/govuk-react/govuk-react/issues/1059

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Ready to be merged

**Screenshots**:

image

image

### CODE REVIEW

**Review by penx — APPROVED** [review]

**Comment by penx** [comment]
This looks great, thankyou!

File: components/footer/src/stories.tsx [hunk 1]
```diff
@@ -46,6 +46,32 @@ WithCopyright.args = {
...
+      </Footer.Link>
+      , except where otherwise stated
+    </span>
+  ),
+};
+
+export const WithNoLicense = Template.bind({});
+
+WithNoLicense.args = {
+  meta: (
```
Comment (penx) [inline components/footer/src/stories.tsx:66]: do we need a custom meta for this story? Should it be a separate story?

Comment (domdomegg) [reply components/footer/src/stories.tsx]: Sure - I've removed it :)

Just thought it looked a bit weird with nothing in the footer at all - but probably makes the story simpler to understand.

image

File: components/footer/src/test.tsx [hunk 2]
```diff
@@ -154,4 +170,12 @@ describe('Footer', () => {
   it('matches with navigation and meta snapshot', () => {
     expect(mount(<FooterWithNavigationAndMeta />)).toMatchSnapshot('FooterWithNavigationAndMeta');
   });
+
+  it('matches with custom licence', () => {
+    expect(mount(<FooterWithCustomLicence />)).toMatchSnapshot('FooterWithCustomLicence');
+  });
```
Comment (penx) [inline components/footer/src/test.tsx:176]: for what it's worth, no need to add any new snapshot tests in as we want to move away from them, as long as they're covered by visual regression (i.e. stories in storybook that are picked up by Chromatic)

**Pull request merged**